libs/shared-web/src/utils/auth-selectors/selectShouldInitializeLaunchDarkly.tsx

```diff
@@ -0,0 +1,25 @@
+import { selectHasUserCompletedSignUp } from './selectHasUserCompletedSignUp';
+import { checkShouldUseLaunchDarkly } from '../checkShouldUseLaunchDarkly';
+import { UserContext } from '../../providers/AuthProvider';
+
+/** Determine if LaunchDarkly should be initialized based on the current auth context */
+export function selectShouldInitializeLaunchDarkly(userCtx: UserContext): boolean {
+  const { isLoggedIn, currentOrganization } = userCtx;
+  // don't show fallback if LaunchDarkly isn't enabled
+  if (!checkShouldUseLaunchDarkly()) {
+    return false;
+  }
+
+  // enable feature flags for unauthenticated areas of the app
+  if (!isLoggedIn) {
+    return true;
+  }
+
+  // allow LD to load when the user is created but still in onboarding
+  if (!selectHasUserCompletedSignUp(userCtx)) {
```

Comment:
Can you please elaborate on why this conditional is required?

Reply:
I'll add more to the code comment as well, but after users provide their name, email, and password, we take them to the `OnboardingForm` where they provide details such as job title, use cases, company name, etc. When they reach this page, `isLoggedIn` is `true`, but they don't have an `organizationId` yet that we can use for org-based feature flags. This was a tricky thing since we otherwise rely on orgId when the user is logged in. We need to make this check so as not to cause it to load endlessly, but still initialize LD in the "anonymous" context

Reply:
Sounds good. Thanks.